lib/iris/coords.py

```diff
@@ -1600,9 +1585,6 @@ def _sanitise_array(self, src, ndmin):
         # Ensure the array has enough dimensions.
         # NB. Returns the *same object* if result.ndim >= ndmin
         result = np.array(result, ndmin=ndmin, copy=False)
-        # We don't need to copy the data, but we do need to have our
-        # own view so we can control the shape, etc.
-        result = result.view()
```

Comment:
My reasoning is that this **ought to be** redundant, now that we no longer support LazyArray-s.

I would generally expect taking a view of an (ordinary) real array to be a redundant operation.
Likewise, our `__init__` codes, and data assignment methods, take the view that a Coord or Cube "owns" it's data, so that should be reasonable.
However, the one place it didn't work was because of assigning array.shape as an in-place modification in aux_factoriy code, e.g. [here](https://github.com/SciTools/iris/pull/2519/files#r114524615).
I think that is highly unexpected, really horrible + shouldn't be done -- so I fixed it.